helix-core/src/main/java/org/apache/helix/task/assigner/AssignableInstance.java

```diff
@@ -113,7 +113,7 @@ private void refreshTotalCapacity() {
     if (resourceCapacity == null) {
       resourceCapacity = new HashMap<>();
       resourceCapacity.put(LiveInstance.InstanceResourceType.TASK_EXEC_THREAD.name(),
-          Integer.toString(TaskConstants.DEFAULT_TASK_THREAD_POOL_SIZE));
+          Integer.toString(_liveInstance.getCurrentTaskThreadPoolSize()));
```

Comment:
Do we do a null check on _liveInstance?

Reply:
There is a null check before _liveInstance assignment, see line 72.